test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -496,6 +501,24 @@ protected void withAggregator(
         }
     }
 
+    protected <T extends AggregationBuilder, V extends InternalAggregation> void verifyOutputFieldNames(T aggregationBuilder, V agg)
+        throws IOException {
+        if (aggregationBuilder.getOutputFieldNames().isEmpty()) {
+            // aggregation does not support output field names yet
+            return;
+        }
+
+        if (agg instanceof NumericMetricsAggregation.MultiValue) {
```

Comment:
Why do we need this `instanceof` check?

Reply:
ok, I made it an `assert`, in future there might be implementations for agg types not based on multi value numeric aggs, but I agree the check was premature and did not force the implementation. In case someone implements `getOutputFieldNames` for another agg type the `assert` will force him/her to fix this test.